### PR TITLE
DOC - Moving Tips and Trick from wiki to Style Guide - added Reading …

### PR DESCRIPTION
**Pull request by vijaysaimutyala**
This changes referring to https://github.com/pandas-dev/pandas/issues/30964 is for removing tips and tricks section page from the [wiki](https://github.com/pandas-dev/pandas/wiki/Tips-&-Tricks-for-pandas-dev) to the [Code Style guide](https://pandas.io/docs/development/code_style.html). 

Since this is my first contribution, I've tried to keep this as simple as possible. @simonjayhawkins mentioned that the conventions for urlopen and ZipFile should be captured under our contributing style guide, testing section. I've seen that there is no Testing section and have created a new section and added these two sections from the Wiki post here. Enabling the Travis-CI and Azure CI pipeline build seemed a bit tough for me. I will try to do the build from my side in my next PR. 

Let me know if any corrections are needed.

### CODE REVIEW

File: doc/source/development/code_style.rst
Comment (simonjayhawkins) [inline doc/source/development/code_style.rst:169]: can you add \*\*Good:\*\* above this for consistency. (Personally, I prefer yes/no over good/bad, but let's leave that for another day)

same for next section

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @vijaysaimutyala generally lgtm.

**Comment by MarcoGorelli** [comment]
Hi @VijayantSoni - is this still active? If so, can you address Simon's comment?

**Comment by vijaysaimutyala** [comment]
Sorry I somehow missed the comment. I'll add this and update here.

**Comment by vijaysaimutyala** [comment]
@MarcoGorelli Updated the PR with comments. Somehow my local build of the docs got pushed. Is there something I should do in future to avoid this or will the local build be ignored ? 

Also, should I be closing the PR if I don't have any further changes on this or will it be done automatically ?

**Comment by MarcoGorelli** [comment]
Thanks for updating, @vijaysaimutyala !

> Somehow my local build of the docs got pushed. Is there something I should do in future to avoid this or will the local build be ignored ?

It's generally safer to stage the files you want to commit with `git add <file>` rather than `git add .`, which can end up with you adding a lot more than you intended. Though to be honest I'm not sure how you ended up with those files, I thought they would be in `doc/_build` (which `git` will ignore) - did you follow the instructions [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#how-to-build-the-pandas-documentation)?

If you could reset any unrelated files to how they are on master, that would be great. IMO the easiest thing at this point would be to reset everything to your first commit:
```
git rm -rf doc/source/_build
git commit -m "<some descriptive commit message>"
git push origin <your branch name>
```
should work, I think.

> Also, should I be closing the PR if I don't have any further changes on this or will it be done automatically ?

If the core team is happy with it, they'll merge it in :)

**Comment by vijaysaimutyala** [comment]
Thanks for the clear answers @MarcoGorelli . Honestly I didn't look at the gitignore but was hoping they would be ignored, but they somehow ended up in my commit. I'll be careful from here on.

I'll try out what you've mentioned. Thanks again 🙂

File: doc/source/development/code_style.rst
Comment (MarcoGorelli) [inline doc/source/development/code_style.rst:163]: >  @simonjayhawkins mentioned that the conventions for urlopen and ZipFile should be captured under our contributing style guide, testing section.

TBH I think that may have been a typo, he was probably copy-and-pasting from his above comment and forgot to change this part, as the two examples you've put below aren't about testing. So, instead of "Testing", I think this section should be called "Misc" (Simon please correct if I'm wrong about this, in which case - my apologies :) )

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @vijaysaimutyala, looks like you've successfully removed the unrelated files

File: doc/source/development/code_style.rst
Comment (vijaysaimutyala) [reply doc/source/development/code_style.rst:163]: I was thinking the same but wasn't sure. I'll update it Misc once @simonjayhawkins confirms.

Comment (simonjayhawkins) [reply doc/source/development/code_style.rst:163]: Misc is fine. testing was incorrect.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @vijaysaimutyala lgtm pending green

**Comment by WillAyd** [comment]
I think we can just delete this wiki entry instead of moving it as its pretty outdated. The second code snippet won't run

**Comment by simonjayhawkins** [comment]
> I think we can just delete this wiki entry instead of moving it as its pretty outdated.

see https://github.com/pandas-dev/pandas/issues/30964#issuecomment-632024140

The content of this PR is not outdated AFAIK, our standard is to use urlopen and ZipFile from pandas.io.common. This PR documents that?

**Comment by simonjayhawkins** [comment]
Travis CI passed but not updated here. will merge shortly if no objections.

**Comment by WillAyd** [comment]
Does ZipFile even exist? I didn’t see it there maybe overlooked. Thought we just used stdlib

**Comment by simonjayhawkins** [comment]
> Does ZipFile even exist? I didn’t see it there maybe overlooked. Thought we just used stdlib
> […](#)
> Sent from my iPhone
> On Jun 23, 2020, at 11:52 AM, Simon Hawkins ***@***.***> wrote: ﻿ Travis CI passed but not updated here. will merge shortly if no objections. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

no it's no longer there. we only have _BytesZipFile now.

@vijaysaimutyala can you remove the ZipFile bit.

@WillAyd OK with urlopen? I think if we maybe add "Lazy-import wrapper for stdlib urlopen, as that imports a big chunk of
    the stdlib." might help?

**Comment by vijaysaimutyala** [comment]
> > Does ZipFile even exist? I didn’t see it there maybe overlooked. Thought we just used stdlib
> > […](#)
> > Sent from my iPhone
> > On Jun 23, 2020, at 11:52 AM, Simon Hawkins _**@**_.***> wrote: ﻿ Travis CI passed but not updated here. will merge shortly if no objections. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> no it's no longer there. we only have _BytesZipFile now.
> 
> @vijaysaimutyala can you remove the ZipFile bit.
> 
> @WillAyd OK with urlopen? I think if we maybe add "Lazy-import wrapper for stdlib urlopen, as that imports a big chunk of
> the stdlib." might help?

Sure. Let me know if you want me to add anything else.

**Comment by simonjayhawkins** [comment]
Thanks @vijaysaimutyala . the CI failures are probably unrelated and fixed on master. can you merge upstream/master to get ci to green to be sure.

**Comment by vijaysaimutyala** [comment]
@simonjayhawkins I merged from upstream and everything looks good. How do I get the CI to run again ?

image

**Comment by MarcoGorelli** [comment]
@vijaysaimutyala if you just push again, the CI process will restart.

(terminology nitpick, but it looks like you did a rebase instead of a merge, in which case you would need to do a forced push)

**Comment by vijaysaimutyala** [comment]
Done :)

**Review by simonjayhawkins — APPROVED** [review]
lgtm @WillAyd

**Comment by WillAyd** [comment]
Thanks @vijaysaimutyala

**Comment by vijaysaimutyala** [comment]
Thanks to you guys for holding my hand throughout. Though this is a small documentation that might not take much time for you, all of you were patient enough for me to understand and move in a proper direction. I use pandas on a daily basis at work and now I'm proud that I too have put in a little effort. Looking forward for big ones 😁

**Pull request merged**
